internal/backend/auth/authhttpmiddleware/authmiddleware.go

```diff
@@ -34,19 +36,25 @@ type Deps struct {
 	Tokens   authtokens.Tokens  `optional:"true"`
 }
 
-func newTokensMiddleware(next http.Handler, tokens authtokens.Tokens) http.HandlerFunc {
+func newTokensMiddleware(next http.Handler, deps Deps) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		ctx := r.Context()
-		ctx = akCtx.WithRequestOrginator(ctx, akCtx.User)
-
-		if user := authcontext.GetAuthnUser(ctx); !user.IsValid() {
-			if auth := r.Header.Get("Authorization"); auth != "" {
-				kind, payload, _ := strings.Cut(auth, " ")
+		ctx := akCtx.WithRequestOrginator(r.Context(), akCtx.User)
+		user := authcontext.GetAuthnUser(ctx)
+		authHdr := r.Header.Get("Authorization")
 
+		if deps.Cfg.AllowDefaultUser {
+			if user.IsValid() || authHdr != "" {
+				http.Error(w, "only default user is allowed", http.StatusUnauthorized)
+				return
+			}
+			ctx = authcontext.SetAuthnUser(ctx, sdktypes.DefaultUser)
+		} else {
+			if !user.IsValid() && authHdr != "" {
```

Comment:
If the user is not valid and the authHeader is empty, I think you can pass the test and continue.

Isn't it problematic?

Auth header has to be there if users are enabled (default user if off)

Reply:
https://linear.app/autokitteh/issue/ENG-1266/auth-middleware-and-no-user